R/zchunk_L2012.ag_For_Past_bio_input_irr_mgmt.R

```diff
@@ -39,133 +43,445 @@ module_aglu_L2012.ag_For_Past_bio_input_irr_mgmt_DISABLED <- function(command, .
+      filter(!region %in% aglu.NO_AGLU_REGIONS) ->
+      L2012.AgSupplySector
+
+    # L2012.AgSupplySubsector: Generic AgSupplySubsector characteristics (none specified as competition is in the land allocator)
+    # At the subsector (production) level, only region x GLU combinations that actually exist are created.
+    # So start with template production tables of available region x commodity x glu for all commodities.
+    # First, biograss: available anywhere that has any crop production at all
+    L103.ag_Prod_Mt_R_C_Y_GLU %>%
+      select(GCAM_region_ID, GLU) %>%
+      unique %>%
+      mutate(GCAM_commodity = "biomass_grass") ->
+      L201.R_C_GLU_biograss
+    # Second, biotree: available anywhere that has any forest production at all
+    L123.For_Prod_bm3_R_Y_GLU %>%
+      select(GCAM_region_ID, GLU) %>%
+      unique %>%
+      mutate(GCAM_commodity = "biomass_tree") ->
+      L201.R_C_GLU_biotree
+    # Third, bind Ag commodties, forest, pasture and biomass all together
+    L103.ag_Prod_Mt_R_C_Y_GLU %>%
+      bind_rows(L123.For_Prod_bm3_R_Y_GLU, L123.ag_Prod_Mt_R_Past_Y_GLU) %>%
+      select(GCAM_region_ID, GCAM_commodity, GLU) %>%
+      unique %>%
+      bind_rows(L201.R_C_GLU_biograss, L201.R_C_GLU_biotree) %>%
+      arrange(GCAM_region_ID, GLU, GCAM_commodity) %>%
+      left_join_error_no_match(A_AgSupplySubsector, by = c("GCAM_commodity" = "AgSupplySubsector")) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      left_join_error_no_match(basin_to_country_mapping[c("GLU_code", "GLU_name")], by = c("GLU" = "GLU_code")) %>%
+      # Subsector isn't just the supplysector & GLU for biomass crops, as this is where the grass/tree split is done
+      mutate(AgSupplySubsector = paste(GCAM_commodity, GLU_name, sep = "_"),
+             # We do not actually care about the logit here but we need a value to avoid errors
+             logit.year.fillout = min(BASE_YEARS), logit.exponent = -3) %>%
+      select(one_of(LEVEL2_DATA_NAMES[["AgSupplySubsector"]])) ->
+      L2012.AgSupplySubsector
+
+    # L2012.AgProduction_ag_irr_mgm: Agricultural commodities production by all technoligies
+    # Start with L2011.AgProduction_ag_irr to get subsector and technology shareweights
+    L161.ag_irrProd_Mt_R_C_Y_GLU %>%
+      mutate(IRR_RFD = "IRR") %>%
+      bind_rows(mutate(L161.ag_rfdProd_Mt_R_C_Y_GLU, IRR_RFD = "RFD")) %>%
+      filter(year %in% BASE_YEARS) %>%
+      mutate(calOutputValue = round(value, digits = aglu.DIGITS_CALOUTPUT)) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      left_join_error_no_match(basin_to_country_mapping[c("GLU_code", "GLU_name")], by = c("GLU" = "GLU_code")) %>%
+      # Set subsector year and technology shareweights, subsector shareweights are set at the aggregate level later
+      mutate(share.weight.year = year,
+             tech.share.weight = if_else(calOutputValue > 0, 1, 0)) ->
```

Comment:
Thanks for pointing this out @abigailsnyder. I'm aware that `replace` is faster than `if_else`. 
When creating new variables, is it worth of doing two lines like: i.e. `tech.share.weight = 0, tech.share.weight = replace(tech.share.weight, calOutputValue > 0, 1)`? Or is there better ways?